### PR TITLE
ansiblegate: take care of failed, skipped and unreachable tasks and propagate "retcode"

### LINKED ISSUE #57854: [BUG] Ansible execution and state modules do not properly report failures
**Description**
When a playbook or ansible.call run is issued and fails, Saltstack doesn't report a failure.

**Setup**
I don't have a great example... I'm testing with Ansible's GCE modules/cloud modules for orchestration purposes...

_Salt state example_
```
{% set name = 'test-instance %}
{% set domain = 'example.lcl' %}
{% set service_account_file = '/path/to/serviceaccount/keyfile.json' %}
{% set project = 'my-gce-project %}
{% set zone = 'us-west1-a' %}
{% set machine_type = 'n1-standard-1' %}
{% set image = 'projects/suse-byos-cloud/global/images/sles-15-sp1-byos-v20200610' %}
{%
    set network_interfaces = [
                               { 'network': { 'selfLink': "projects/my-gce-project/global/networks/default" },
                                 'subnetwork': { 'selfLink': "projects/my-gce-project/regions/us-west1/subnetworks/default" } },
                             ]
%}

{% set ansible_kwargs = {'extra_vars':
                            { 'name': name,
                              'domain': domain,
                              'service_account_file': service_account_file,
                              'project': project,
                              'zone': zone,
                              'machine_type': machine_type,
                              'image': image,
                              'network_interfaces': network_interfaces

                            }
       }
%}

orch_instance_gcp_deployed:
    module.run:
        - name: ansible.playbooks
        - playbook: /srv/salt/orch/instance/gcp/playbooks/playbook-deployed.yml
        - check: False
        - extra_vars: {{ ansible_kwargs['extra_vars'] | tojson }}
```
_ansible playbook example:_
```
- name: Create GCP Instance
  hosts: localhost
  gather_facts: no

  tasks:
  - name: Create a GCP Instance
    gcp_compute_instance:
        name: "{{ name }}"
        hostname: "{{ name }}.{{ domain }}"
        machine_type: "{{ machine_type }}"
        disks:
            - auto_delete: True
              boot: yes
              initialize_params:
                  source_image: "{{ image }}"
                  disk_size_gb: 40
        network_interfaces: "{{ network_interfaces }}"
        zone: "{{ zone }}"
        project: "{{ project }}"
        auth_kind: serviceaccount
        service_account_file: "{{ service_account_file }}"
        state: present
```
**Steps to Reproduce the behavior**
If the above state is run twice, it will produce an error (since the instance already exists). Salt will mark it as successful with changes. 

If you choose to use the above example, you'll need to create a gcp service account in a gcp project and acquire a service account key. https://cloud.google.com/iam/docs/creating-managing-service-account-keys
**Expected behavior**
Salt should report it as failed.

**Screenshots**
_Saltstack output:_
```
module_|-orch_instance_gcp_deployed_|-ansible.playbooks_|-run:
    ----------
    __id__:
        orch_instance_gcp_deployed
    __run_num__:
        0
    __sls__:
        orch.instance.gcp.deployed
    changes:
        ----------
        ret:
            ----------
            custom_stats:
                ----------
            global_custom_stats:
                ----------
            plays:
                |_
                  ----------
                  play:
                      ----------
                      duration:
                          ----------
                          end:
                              2020-07-02T15:11:07.821863Z
                          start:
                              2020-07-02T15:11:06.566000Z
                      id:
                          42010a20-0458-ae8b-0d4c-000000000005
                      name:
                          Create GCP Instance
                  tasks:
                      |_
                        ----------
                        hosts:
                            ----------
                            localhost:
                                ----------
                                _ansible_no_log:
                                    False
                                action:
                                    gcp_compute_instance
                                changed:
                                    False
                                exception:
                                    Traceback (most recent call last):
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 102, in <module>
                                        _ansiballz_main()
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 94, in _ansiballz_main
                                        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 40, in invoke_module
                                        runpy.run_module(mod_name='ansible.modules.cloud.google.gcp_compute_instance', init_globals=None, run_name='__main__', alter_sys=True)
                                      File "/usr/lib64/python3.6/runpy.py", line 205, in run_module
                                        return _run_module_code(code, init_globals, run_name, mod_spec)
                                      File "/usr/lib64/python3.6/runpy.py", line 96, in _run_module_code
                                        mod_name, mod_spec, pkg_name, script_name)
                                      File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
                                        exec(code, run_globals)
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1739, in <module>
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1062, in main
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1092, in update
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1098, in update_fields
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1383, in deletion_protection_update
                                    KeyError: 'deletionProtection'
                                failed:
                                    True
                                module_stderr:
                                    Traceback (most recent call last):
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 102, in <module>
                                        _ansiballz_main()
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 94, in _ansiballz_main
                                        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
                                      File "/var/lib/salt/.ansible/tmp/ansible-tmp-1593702666.6738384-145552361200225/AnsiballZ_gcp_compute_instance.py", line 40, in invoke_module
                                        runpy.run_module(mod_name='ansible.modules.cloud.google.gcp_compute_instance', init_globals=None, run_name='__main__', alter_sys=True)
                                      File "/usr/lib64/python3.6/runpy.py", line 205, in run_module
                                        return _run_module_code(code, init_globals, run_name, mod_spec)
                                      File "/usr/lib64/python3.6/runpy.py", line 96, in _run_module_code
                                        mod_name, mod_spec, pkg_name, script_name)
                                      File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
                                        exec(code, run_globals)
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1739, in <module>
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1062, in main
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1092, in update
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1098, in update_fields
                                      File "/tmp/ansible_gcp_compute_instance_payload_ojdpx3jt/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py", line 1383, in deletion_protection_update
                                    KeyError: 'deletionProtection'
                                module_stdout:
                                msg:
                                    MODULE FAILURE
                                    See stdout/stderr for the exact error
                                rc:
                                    1
                        task:
                            ----------
                            duration:
                                ----------
                                end:
                                    2020-07-02T15:11:07.821863Z
                                start:
                                    2020-07-02T15:11:06.587196Z
                            id:
                                42010a20-0458-ae8b-0d4c-000000000007
                            name:
                                Create a GCP Instance
            stats:
                ----------
                localhost:
                    ----------
                    changed:
                        0
                    failures:
                        1
                    ignored:
                        0
                    ok:
                        0
                    rescued:
                        0
                    skipped:
                        0
                    unreachable:
                        0
    comment:
        Module function ansible.playbooks executed
    duration:
        3034.303
    name:
        ansible.playbooks
    result:
        True
    start_time:
        09:11:04.952419
```
_Ansible output:_
```
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: Found variable using reserved name: name
 ____________________________
< PLAY [Create GCP Instance] >
 ----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

 ______________________________
< TASK [Create a GCP Instance] >
 ------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'deletionProtection'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1593722799.1080034-11932906934944/AnsiballZ_gcp_compute_instance.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1593722799.1080034-11932906934944/AnsiballZ_gcp_compute_instance.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1593722799.1080034-11932906934944/AnsiballZ_gcp_compute_instance.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.cloud.google.gcp_compute_instance', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_gcp_compute_instance_payload_cjoahokh/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py\", line 1739, in <module>\n  File \"/tmp/ansible_gcp_compute_instance_payload_cjoahokh/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py\", line 1062, in main\n  File \"/tmp/ansible_gcp_compute_instance_payload_cjoahokh/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py\", line 1092, in update\n  File \"/tmp/ansible_gcp_compute_instance_payload_cjoahokh/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py\", line 1098, in update_fields\n  File \"/tmp/ansible_gcp_compute_instance_payload_cjoahokh/ansible_gcp_compute_instance_payload.zip/ansible/modules/cloud/google/gcp_compute_instance.py\", line 1383, in deletion_protection_update\nKeyError: 'deletionProtection'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```
**Versions Report**
I'm running 2019.2 but I've tested with the latest ansible execution and state modules in my environment and am having the same problems. The goal is to use this as a part of an orchestration so I'm executing the state via salt-run salt.cmd state.apply orch.instance.gcp.deployed.
<details><summary>
Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: 1.11.2
       cherrypy: unknown
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: 1.6.1
       M2Crypto: 0.35.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.10 (default, Dec 19 2019, 15:48:40) [GCC]
   python-gnupg: Not Installed
         PyYAML: 5.1.2
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3

System Versions:
           dist:
         locale: UTF-8
        machine: x86_64
        release: 4.12.14-197.45-default
         system: Linux
        version: Not Installed</summary>
</details>

### LINKED ISSUE #57855: [BUG] ansible.playbooks state module doesn't work properly with modules that do not support checks
**Description**
When the ansible.playbooks state module is used, the --check flag is enforced and cannot be disabled. If the ansible module executed does not support checks, the task will be skipped.

**Setup**
I don't have a great example... I'm testing with Ansible's GCE modules/cloud modules for orchestration purposes...

_Salt state example_
```
{% set name = 'test-instance %}
{% set domain = 'example.lcl' %}
{% set service_account_file = '/path/to/serviceaccount/keyfile.json' %}
{% set project = 'my-gce-project %}
{% set zone = 'us-west1-a' %}
{% set machine_type = 'n1-standard-1' %}
{% set image = 'projects/suse-byos-cloud/global/images/sles-15-sp1-byos-v20200610' %}
{%
    set network_interfaces = [
                               { 'network': { 'selfLink': "projects/my-gce-project/global/networks/default" },
                                 'subnetwork': { 'selfLink': "projects/my-gce-project/regions/us-west1/subnetworks/default" } },
                             ]
%}

{% set ansible_kwargs = {'extra_vars':
                            { 'name': name,
                              'domain': domain,
                              'service_account_file': service_account_file,
                              'project': project,
                              'zone': zone,
                              'machine_type': machine_type,
                              'image': image,
                              'network_interfaces': network_interfaces

                            }
       }
%}

orch_instance_gcp_deployed:
    ansible.playbooks:
        - name: /srv/salt/orch/instance/gcp/playbooks/playbook-deployed.yml
        - ansible_kwargs: {{ ansible_kwargs | tojson }}
```
_ansible playbook example:_
```
- name: Create GCP Instance
  hosts: localhost
  gather_facts: no

  tasks:
  - name: Create a GCP Instance
    gcp_compute_instance:
        name: "{{ name }}"
        hostname: "{{ name }}.{{ domain }}"
        machine_type: "{{ machine_type }}"
        disks:
            - auto_delete: True
              boot: yes
              initialize_params:
                  source_image: "{{ image }}"
                  disk_size_gb: 40
        network_interfaces: "{{ network_interfaces }}"
        zone: "{{ zone }}"
        project: "{{ project }}"
        auth_kind: serviceaccount
        service_account_file: "{{ service_account_file }}"
        state: present
```
**Steps to Reproduce the behavior**
This should be reproducable with a different example. I'm not familiar enough with ansible to know what modules don't support --check. I assume most of their cloud orchestration stuff wouldn't support --check.

If you choose to use the above example, you'll need to create a gcp service account in a gcp project and acquire a service account key. https://cloud.google.com/iam/docs/creating-managing-service-account-keys

**Expected behavior**
Salt should report it as failed.

**Screenshots**
_If run with salt:_
```
ansible_|-orch_instance_gcp_deployed_|-/srv/salt/orch/instance/gcp/playbooks/playbook-deployed.yml_|-playbooks:
    ----------
    __id__:
        orch_instance_gcp_deployed
    __run_num__:
        0
    __sls__:
        orch.instance.gcp.deployed
    changes:
        ----------
    comment:
        No changes to be made from playbook /srv/salt/orch/instance/gcp/playbooks/playbook-deployed.yml
    duration:
        3995.633
    name:
        /srv/salt/orch/instance/gcp/playbooks/playbook-deployed.yml
    result:
        True
    start_time:
        09:31:06.519698
```
_if run with ansible (with --check and --diff flags):_
```
Using /etc/ansible/ansible.cfg as config file
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: Found variable using reserved name: name
 ____________________________
< PLAY [Create GCP Instance] >
 ----------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

 ______________________________
< TASK [Create a GCP Instance] >
 ------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

skipping: [localhost] => {"changed": false, "msg": "remote module (gcp_compute_instance) does not support check mode"}
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

localhost                  : ok=0    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```
**Expected behavior**
Salt should either allow one to override the check flag in the ansible.playbooks state module or perhaps disable --check entirely (not sure if --check is required....)

**Versions Report**
I'm running 2019.2 but I've tested with the latest ansible execution and state modules in my environment and am having the same problems. The goal is to use this as a part of an orchestration so I'm executing the state via salt-run salt.cmd state.apply orch.instance.gcp.deployed.
<details><summary>
Salt Version:
           Salt: 2019.2.0

Dependency Versions:
           cffi: 1.11.2
       cherrypy: unknown
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
        libnacl: 1.6.1
       M2Crypto: 0.35.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.10 (default, Dec 19 2019, 15:48:40) [GCC]
   python-gnupg: Not Installed
         PyYAML: 5.1.2
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3

System Versions:
           dist:
         locale: UTF-8
        machine: x86_64
        release: 4.12.14-197.45-default
         system: Linux
        version: Not Installed</summary>
</details>

### PR DESCRIPTION
**Pull request by meaksh**
### What does this PR do?
This PR fixes few issues on the `ansiblegate` execution and state modules when running playbooks:

- Take care of "failed", "skipped" and "unreachable" tasks during playbook execution.
- Propagate "retcode" from "ansible-playbook" CLI execution when calling `ansible.playbooks` to allow proper success/failed detection.
- Only execute `ansible-playbook --check` in test mode (`test=True`) to allow running playbooks that are using modules which does not allow `--check`.
- Add new unit tests for execution and state modules.  
- Add changes suggested by "pre-commit".

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/57854
Fixes: https://github.com/saltstack/salt/issues/57855

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/unit/states/test_ansiblegate.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from tests.support.unit import TestCase, skipIf
+
+try:
+    import pytest
+except ImportError as import_error:
+    pytest = None
+NO_PYTEST = not bool(pytest)
+
+
+@skipIf(NO_PYTEST, False)
```
Comment (DmitryKuzmenko) [inline tests/unit/states/test_ansiblegate.py:35]: Just cosmetic request to drop the redundant `NO_PYTEST` constant. Other tests are using `pytest` directly in `skipIf`
```python
@skipIf(pytest is None, "PyTest is missing")
```

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Comment by DmitryKuzmenko** [comment]
@meaksh thank you for contribution. Could you please follow up just one cosmetic change request?

File: tests/unit/states/test_ansiblegate.py (hunk 1 above)
Comment (meaksh) [reply tests/unit/states/test_ansiblegate.py]: :+1:

**Comment by meaksh** [comment]
@DmitryKuzmenko I've pushed the changes you requested. Thanks for reviewing! :+1:

File: tests/pytests/unit/modules/test_ansiblegate.py [hunk 2]
```diff
@@ -131,3 +131,75 @@ def test_virtual_function(resolver):
...
+                        "echo",
+                        '{"ANSIBLE_MODULE_ARGS": {"_raw_params": "arg_1", "kwarg1": "foobar"}}',
+                    ],
+                    stdout=-1,
+                    timeout=1200,
+                )
+            assert ret == {"completed": True, "timeout": 1200}
+
+
+@patch("salt.utils.path.which", MagicMock(return_value=True))
```
Comment (DmitryKuzmenko) [inline tests/pytests/unit/modules/test_ansiblegate.py:195]: Could you please move this mock object into the function level scope. Having mocks in the module level scope could produce issues.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_ansiblegate.py (hunk 2 above)
Comment (meaksh) [reply tests/pytests/unit/modules/test_ansiblegate.py:195]: Ok, good to know, thanks! I've pushed the requested changes :+1:

Comment (meaksh) [reply tests/pytests/unit/modules/test_ansiblegate.py:195]: Oh wait, I missed few of them. 1 sec.

Comment (meaksh) [reply tests/pytests/unit/modules/test_ansiblegate.py]: Ok, it should be ready now.

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by DmitryKuzmenko** [comment]
Looks good now. Thank you for update.
Let's wait for CI results.

**Comment by denis-khalturin-incountry** [comment]
@meaksh why are you marking all skipped tasks as outstanding? this behavior terminates the application of the state with an error

**Comment by meaksh** [comment]
Hey @denis-khalturin-incountry, I don't really get what you mean. Could you please elaborate that and provide some examples?

Thanks in advance!

**Comment by denis-khalturin-incountry** [comment]
@meaksh yes of course.

i am applying playbook:
`ansible-playbook playbooks/test.yml --diff --extra-vars='{"ansible_connection": "local", "inventory_hostname": "_my_host_"}' --inventory=inventory --forks = 5
`

One of the tasks is skipped according to the conditions:
`PLAY RECAP *********************************************** ************************************************* ************************************************* ************************************************* *************************
_my_host_: ok = 7 changed = 0 unreachable = 0 failed = 0 skipped = 1 rescued = 0 ignored = 0`

But if I start the playbook using the state, I get a runtime error:
`
salt-call -l debug state.apply
....
[INFO    ] Executing command 'ansible-playbook playbooks/test.yml --diff --extra-vars='{"ansible_connection": "local", "inventory_hostname": "_my_host_"}' --inventory=inventory --forks=5' in directory 'ansible'
....
[ERROR   ] {'all:!without_hardening': {'_my_role_ : _task_that_is_skipped_':
....
[INFO    ] Completed state [playbooks/test.yml] at time 11:51:25.176235 (duration_in_ms=6288.308)
....
----------
          ID: ansible playbook
    Function: ansible.playbooks
        Name: playbooks/test.yml
      Result: False
     Comment: There were some issues running the playbook playbooks/test.yml
     Started: 11:51:18.887927
    Duration: 6288.308 ms
     Changes:
....
Succeeded: 6 (changed=2)
Failed:    1
`

I think this is the wrong handling of skipped tasks

**Comment by meaksh** [comment]
Thanks for clarifying @denis-khalturin-incountry ! You're totally right :+1:

I did some checks here and can confirm that `ansible-playbook` CLI does not set "retcode != 0" when there are skipped tasks. Therefore we should not force the Salt state to fail in such cases.

Could you please create a new issue so we can hopefully produce a fix soon?

Thanks in advance!

**Comment by denis-khalturin-incountry** [comment]
Yes, I will create an issue. Thanks

**Pull request merged**